dockerd-rootless-setuptool.sh: add `nsenter` subcommand (for debugging)

AkihiroSuda opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/moby/moby/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

For additional information on our contributing process, read our contributing
guide https://docs.docker.com/opensource/code/

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**
Added `nsenter` subcommand to `dockerd-rootless-setuptool.sh`.

Expected to be used for debugging.

**- How I did it**
Ported `containerd-rootless-setuptool.sh nsenter` (part of nerdctl) to `dockerd-rootless-setuptool.sh`
https://github.com/containerd/nerdctl/blob/v1.7.2/extras/rootless/containerd-rootless-setuptool.sh#L142-L147


**- How to verify it**

```console
$ dockerd-rootless-setuptool.sh nsenter -- ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: tap0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 65520 qdisc pfifo_fast state UP group default qlen 1000
    link/ether ae:c3:a3:20:53:3e brd ff:ff:ff:ff:ff:ff
    inet 10.0.2.100/24 scope global tap0
       valid_lft forever preferred_lft forever
    inet6 fe80::acc3:a3ff:fe20:533e/64 scope link 
       valid_lft forever preferred_lft forever
3: docker0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN group default 
    link/ether 02:42:a6:36:b0:a9 brd ff:ff:ff:ff:ff:ff
    inet 172.17.0.1/16 brd 172.17.255.255 scope global docker0
       valid_lft forever preferred_lft forever
```

```console
$ dockerd-rootless-setuptool.sh nsenter -- cat /proc/self/uid_map 
         0       1001          1
         1     100000      65536
     65537     200000      65536
```

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->
dockerd-rootless-setuptool.sh: add `nsenter` subcommand (for debugging)

**- A picture of a cute animal (not mandatory but encouraged)**
🐧

AkihiroSuda committed:
dockerd-rootless.sh: set `rootlesskit --state-dir=DIR`

Now the state dir is set to `${XDG_RUNTIME_DIR}/dockerd-rootless`.

This is similar to `${XDG_RUNTIME_DIR}/containerd-rootless` used in nerdctl:
https://github.com/containerd/nerdctl/blob/v1.7.2/extras/rootless/containerd-rootless.sh#L35

Prior to this commit, the state dir was unset and a random dir under `/tmp` was used.
(e.g., `/tmp/rootlesskit1869901982`)

Signed-off-by: Akihiro Suda <akihiro.suda.cz@hco.ntt.co.jp>

AkihiroSuda committed:
dockerd-rootless-setuptool.sh: add `nsenter` subcommand

Usage: `dockerd-rootless-setuptool.sh nsenter -- ip a`

Expected to be used for debugging.

Ported from nerdctl's `containerd-rootless-setuptool.sh`
https://github.com/containerd/nerdctl/blob/v1.7.2/extras/rootless/containerd-rootless-setuptool.sh#L142-L147

Signed-off-by: Akihiro Suda <akihiro.suda.cz@hco.ntt.co.jp>

thaJeztah commented:
Some failures; not sure if related in any way; perhaps you could try a rebase to get a fresh run?

AkihiroSuda committed:
testutil/daemon: shorten XDG_RUNTIME_DIR

XDG_RUNTIME_DIR will contain sockets so its path mustn't be too long.
Prior to this commit, it was set to very long path like
`/go/src/github.com/docker/docker/bundles/test-integration/TestDiskUsage/de4fb36576d7d/xdgrun`

Signed-off-by: Akihiro Suda <akihiro.suda.cz@hco.ntt.co.jp>

AkihiroSuda commented:
Now CI is green

thaJeztah commented:
Arf.. those "too long" paths keep on coming back. Perhaps something we should look into at some point (maybe use some short temp-dir somewhere for test-daemons to put the socket in).

In either case, that's a separate concern, good for handling separate / follow up work.

thaJeztah reviewed (approved):
LGTM
